### PR TITLE
Replace Parallelize with function ParallelizeUntil and formally depre…

### PR DESCRIPTION
**Pull request by wgliang**
…cate the Parallelize

**What this PR does / why we need it**:
And formally deprecate the Parallelize(It's just a variant of ParallelizeUntil).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
This is a follow up PR of #67555 
FYI https://github.com/kubernetes/kubernetes/pull/67555#discussion_r212155466

**Release note**:

```release-note
Replace Parallelize with function ParallelizeUntil and formally deprecate the Parallelize.
```

### CODE REVIEW

File: staging/src/k8s.io/client-go/util/workqueue/parallelizer.go [hunk 1]
```diff
@@ -25,12 +25,6 @@ import (
 
 type DoWorkPieceFunc func(piece int)
 
-// Parallelize is a very simple framework that allows for parallelizing
-// N independent pieces of work.
-func Parallelize(workers, pieces int, doWorkPiece DoWorkPieceFunc) {
```
Comment (wgliang) [inline staging/src/k8s.io/client-go/util/workqueue/parallelizer.go]: Delete or just note Deprecate?

**Comment by wojtek-t** [comment]
> Will we merge in v.1.12?

No - it's code-freeze already, and it's not a bug fix or anything like that.

**Comment by wgliang** [comment]
@wojtek-t 
Ok, make sense.

File: pkg/scheduler/algorithm/predicates/metadata.go [hunk 2]
```diff
@@ -421,7 +421,7 @@ func getPodsMatchingAffinity(pod *v1.Pod, nodeInfoMap map[string]*schedulercache
 			appendResult(node.Name, nodeTopologyPairsAffinityPodsMaps, nodeTopologyPairsAntiAffinityPodsMaps)
 		}
 	}
-	workqueue.Parallelize(16, len(allNodeNames), processNode)
+	workqueue.ParallelizeUntil(nil, 16, len(allNodeNames), processNode)
```
Comment (sttts) [inline pkg/scheduler/algorithm/predicates/metadata.go]: is a nil context an anti-pattern? Never seen that. There is `context.Background()` and `context.TODO()`.

Comment (sttts) [reply pkg/scheduler/algorithm/predicates/metadata.go]: these cases in the scheduler all look like one of those two.

File: staging/src/k8s.io/client-go/util/workqueue/parallelizer.go (hunk 1 above)
Comment (sttts) [reply staging/src/k8s.io/client-go/util/workqueue/parallelizer.go]: deprecate. We might have users of it.

File: pkg/scheduler/algorithm/predicates/metadata.go (hunk 2 above)
Comment (wgliang) [reply pkg/scheduler/algorithm/predicates/metadata.go]: yeah, you are right. The default will be context.TODO(), but not nil. But ParallelizeUntil will handle it when the context is nil.

File: staging/src/k8s.io/client-go/util/workqueue/parallelizer.go (hunk 1 above)
Comment (wgliang) [reply staging/src/k8s.io/client-go/util/workqueue/parallelizer.go]: Make sense.

**Comment by wgliang** [comment]
@sttts I've updated it. But do we need to change the `nil` in workqueue.Parallelize to the default `context.TODO()`
https://github.com/kubernetes/kubernetes/blob/426ef9d349bb3a277c3e8826fb772b6bdb008382/staging/src/k8s.io/client-go/util/workqueue/parallelizer.go#L31

**Review by tao12345666333 — APPROVED** [review]

**Comment by wgliang** [comment]
/ping @sttts 
for approve

File: staging/src/k8s.io/client-go/util/workqueue/parallelizer.go [hunk 3]
```diff
@@ -25,6 +25,9 @@ import (
 
 type DoWorkPieceFunc func(piece int)
 
+// DEPRECATED: Parallelize is deprecated. ParallelizeUntil is a more
```
Comment (sttts) [inline staging/src/k8s.io/client-go/util/workqueue/parallelizer.go]: this should be in the default deprecation style, compare https://rakyll.org/deprecated/

Comment (wgliang) [reply staging/src/k8s.io/client-go/util/workqueue/parallelizer.go]: DONE. Thanks.

**Review by yastij — APPROVED** [review]
/lgtm

Thanks @wgliang !

**Pull request merged**